url-updater/src/main/java/com/devonfw/tools/ide/url/updater/AbstractUrlUpdater.java

```diff
@@ -190,58 +201,59 @@ protected boolean isSuccess(HttpResponse<?> response) {
   }
 
   /**
-   * Checks the download URL and takes the provided checksum into account instead of downloading the file and generating
-   * the checksum from it
+   * Checks if the download file checksum is still valid
    *
-   * @param url the URL of the download to check.
-   * @param urlVersion the {@link UrlVersion} where to store the collected information like status and checksum.
+   * @param url String of the URL to check
+   * @param urlVersion String of the version
    * @param os the {@link OperatingSystem}
    * @param architecture the {@link SystemArchitecture}
-   * @param checksum String of the checksum to use
-   * @return {@code true} if the download was checked successfully, {@code false} otherwise.
+   * @param checksum String of the new checksum to check
+   * @param tool String of the tool
+   * @param version String of the version
+   * @return {@code true} if update of checksum was successful, {@code false} otherwise.
    */
-  private boolean checkDownloadUrl(String url, UrlVersion urlVersion, OperatingSystem os,
-      SystemArchitecture architecture, String checksum) {
-
-    HttpResponse<?> response = doCheckDownloadViaHeadRequest(url);
-    int statusCode = response.statusCode();
-    boolean success = isSuccess(response);
-    String tool = getToolWithEdition();
-    String version = urlVersion.getName();
-    String contentType = response.headers().firstValue("content-type").orElse("undefined");
-    if (success && contentType.startsWith("text")) {
-      logger.error("For tool {} and version {} the download has an invalid content type {} for URL {}", tool, version,
-          contentType, url);
-      success = false;
-    }
-    if (success) {
-      UrlDownloadFile urlDownloadFile = urlVersion.getOrCreateUrls(os, architecture);
-      UrlChecksum urlChecksum = urlVersion.getOrCreateChecksum(urlDownloadFile.getName());
+  private static boolean isChecksumStillValid(String url, UrlVersion urlVersion, OperatingSystem os,
+      SystemArchitecture architecture, String checksum, String tool, String version) {
+
+    UrlDownloadFile urlDownloadFile = urlVersion.getOrCreateUrls(os, architecture);
+    UrlChecksum urlChecksum = urlVersion.getOrCreateChecksum(urlDownloadFile.getName());
+    urlDownloadFile.addUrl(url);
+    urlChecksum.setChecksum(checksum);
+    String oldChecksum = urlChecksum.getChecksum();
+
+    if ((oldChecksum != null) && !Objects.equal(oldChecksum, checksum)) {
```

Comment:
IMHO this code does not make sense:
The getter will return what has been given to the setter so `checksum` and `oldChecksum` will always refer to the same value.
IMHO the entire method `isChecksumStillValid` does not make sense yet. If we would have a method with that name then either to verify the checksum for the end-user but that code is currently still in bash or in case we would revalidate this on update of existing versions. However, due to the greediness of checksum computation this is not planned.

Reply:
I've removed the addUrl and setChecksum now. Not sure if we should remove the whole method though.

Reply:
We can discuss this after merge. However, the method still seems pointless to me: if it is called from `addVersion` the version is new and no previous checksum can exist. If we use this method on update of versions this makes sense but in such case we never set the checksum, as we already assume that it cannot change afterwards - a release by design has to be immutable.